arrow/arrow.py

```diff
@@ -904,7 +904,13 @@ def _get_iteration_params(cls, end, limit):
     def _get_timestamp_from_input(timestamp):
 
         try:
-            return float(timestamp)
+            timestamp = float(timestamp)
+            if timestamp >= 253402300800000: # nanoseconds
+                return timestamp / 1000000
+            elif timestamp >= 253402300800: # milliseconds
```

Comment:
1000 is ambiguous, no doubt.
However anything above 253402300800 (which corresponds to 1/1/10000), datetime is not designed to handle. Therefore, anything above that can't be a timestamp in seconds, so arrow could assume it's not rather than throwing an error.